src/vs/workbench/contrib/files/browser/views/explorerViewer.ts

```diff
@@ -1346,36 +1355,58 @@ export class FileDragAndDrop implements ITreeDragAndDrop<ExplorerItem> {
 		return this.workspaceEditingService.updateFolders(0, workspaceCreationData.length, workspaceCreationData);
 	}
 
-	private async doHandleExplorerDrop(source: ExplorerItem, target: ExplorerItem, isCopy: boolean): Promise<void> {
+	private async doHandleExplorerDrop(sources: ExplorerItem[], target: ExplorerItem, isCopy: boolean): Promise<void> {
 		// Reuse duplicate action if user copies
+		const sourceTargetPairs: { source: URI; target: URI }[] = [];
```

Comment:
Maybe doHandleExplorerDrop would be easier to understand if it would split between code for copy operation vs move operation.

Reply:
This is an optional change which I can tackle also once we merge this PR

Reply:
I'll see what I can do before you go on vacation :)

Reply:
I've split into two separate functions.